internal/controller/gateway.go

```diff
@@ -31,46 +31,107 @@ import (
 )
 
 type gatewayReconciler struct {
+	ctx          context.Context
 	client       client.Client
 	eventHandler cache.ResourceEventHandler
-	logrus.FieldLogger
+	log          logrus.FieldLogger
+	// controllerName is the configured controller of managed gatewayclasses.
+	controllerName string
+	// referencedClass is the gatewayclass referenced by managed gateways.
+	referencedClass *gatewayapi_v1alpha1.GatewayClass
 }
 
 // NewGatewayController creates the gateway controller from mgr. The controller will be pre-configured
-// to watch for Gateway objects across all namespaces.
-func NewGatewayController(mgr manager.Manager, eventHandler cache.ResourceEventHandler, log logrus.FieldLogger) (controller.Controller, error) {
+// to watch for Gateway objects across all namespaces and reconcile those that match class.
+func NewGatewayController(mgr manager.Manager, eventHandler cache.ResourceEventHandler, log logrus.FieldLogger, class string) (controller.Controller, error) {
```

Comment:
I think:
```suggestion
func NewGatewayController(mgr manager.Manager, eventHandler cache.ResourceEventHandler, log logrus.FieldLogger, controllerName string) (controller.Controller, error) {
```

Reply:
Yup, I renamed them backwards to the call chain to make it more specific to the reader.